src/components/app/list-new/HelmAppList.tsx

```diff
@@ -270,93 +257,51 @@ export default function HelmAppList({
-            })
-        }
-
-        // handle search
-        if (_search?.length) {
-            _filteredHelmAppsList = _filteredHelmAppsList.filter(
-                (app) =>
-                    app.appName.toLowerCase().includes(_search.toLowerCase()) ||
-                    app.chartName.toLowerCase().includes(_search.toLowerCase()),
+    function _getExternalHelmApps() {
+        setFetchingExternalAppsState(true)
+        const _sseConnection = new EventSource(`${Host}/application?clusterIds=${clusterIdsCsv}`, {
+            withCredentials: true,
+        })
+        const _externalAppReceivedClusterIds = []
+        const _externalAppReceivedHelmApps = []
+        const _externalAppFetchErrors: string[] = []
+        _sseConnection.onmessage = function (message) {
+            _onExternalAppDataFromSse(
+                message,
+                _externalAppReceivedClusterIds,
+                _externalAppReceivedHelmApps,
+                _externalAppFetchErrors,
+                _sseConnection,
             )
         }
-
-        const dynamicSortBy = AppListColumnSort[_sortBy]
-
-        // handle sort
-        if (_sortOrder == OrderBy.ASC) {
-            _filteredHelmAppsList = _filteredHelmAppsList.sort((a, b) =>
-                a[dynamicSortBy].localeCompare(b[dynamicSortBy]),
-            )
-        } else {
-            _filteredHelmAppsList = _filteredHelmAppsList.sort((a, b) =>
-                b[dynamicSortBy].localeCompare(a[dynamicSortBy]),
-            )
+        _sseConnection.onerror = function (err) {
+            _externalAppFetchErrors.push(EXTERNAL_HELM_SSE_CONNECTION_ERROR)
+            setExternalHelmListFetchErrors([..._externalAppFetchErrors])
+            _closeSseConnection(_sseConnection)
         }
-        setSortBy(_sortBy)
-        setSortOrder(_sortOrder)
-        setFilteredHelmAppsList(_filteredHelmAppsList)
-        setShowPulsatingDotState(_filteredHelmAppsList.length == 0 && !clusterIdsCsv)
+        setSseConnection(_sseConnection)
     }
 
     function _isAnyFilterationAppliedExceptClusterAndNs() {
-        return (
-            payloadParsedFromUrl.teams?.length ||
-            payloadParsedFromUrl.appNameSearch?.length ||
-            payloadParsedFromUrl.environments?.length
-        )
+        return project.length || searchKey.length || environment.length || appStatus.length
     }
 
     function _isAnyFilterationApplied() {
-        return _isAnyFilterationAppliedExceptClusterAndNs() || payloadParsedFromUrl.namespaces?.length
+        return _isAnyFilterationAppliedExceptClusterAndNs() || cluster.length || namespace.length
     }
 
     function _isOnlyAllClusterFilterationApplied() {
-        const _isAllClusterSelected = !masterFilters.clusters.some((_cluster) => !_cluster.isChecked)
-        const _isAnyNamespaceSelected = masterFilters.namespaces.some((_namespace) => _namespace.isChecked)
+        const _isAllClusterSelected = cluster.length === clusterList?.length
```

Comment:
Check if ? required

Reply:
required

Reply:
Can we send only the length from parent?